azurelinuxagent/ga/update.py

```diff
@@ -982,6 +986,34 @@ def _ensure_firewall_rules_persisted(dst_ip):
             message=msg,
             log_event=False)
 
+    @staticmethod
+    def _add_dns_tcp_firewall_rule_if_not_enabled(dst_ip):
+
+        # Helper to execute a run command, returns True if no exception else returns False
+        def _execute_run_command(command):
+            try:
+                shellutil.run_command(command)
+                return True
+            except Exception:
+                return False
+
+        # "-C" checks if the iptable rule is available in the chain. It throws an exception if the ip table rule doesnt exist
+        drop_rule = _get_firewall_drop_command("-w", "-C", dst_ip)
```

Comment:
There's an existing function to check weather to add wait or not based on the iptable version (check `wait = self.get_firewall_will_wait()` function). Please use that instead of adding `-w` directly

Reply:
Additionally, can you add a parameter of `-C` too (similar to `-A-`, `-D`, `-I`) and use that instead of using it directly

Reply:
Addressed in the next commit